service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/ControlledAzureVmResource.java

```diff
@@ -88,6 +90,14 @@ public ControlledAzureVmResource(DbResource dbResource) {
     validate();
   }
 
+  /** {@inheritDoc} */
+  @Override
+  public Optional<UniquenessCheckParameters> getUniquenessCheckParameters() {
+    return Optional.of(
+        new UniquenessCheckParameters(UniquenessScope.WORKSPACE)
+            .addParameter("vmName", getVmName()));
```

Comment:
vm name is globally unique, like bucket names?

Reply:
I just transcribed the existing code. It could be it is wrong.